File: LeetCode_Solutions/2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()
        n=len(skill)-1
        tot=skill[0]+skill[n]
       # print(tot)
        chm=0
        for i in range(len(skill)//2):
            if(skill[i]+skill[n-i]==tot):
                chm+=skill[i]*skill[n-i]
             #   print(chm)
            else:
                return -1
        return chm

File: LeetCode_Solutions/2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.java
class Solution {
    public long dividePlayers(int[] skill) {
      Arrays.sort(skill);
      int n=skill.length-1;
      int tot=skill[0]+skill[n];
      long chm=0;
      for(int i=0;i<(n+1)/2;i++){
        if(skill[i]+skill[n-i]==tot){
            chm+=skill[i]*skill[n-i];
        }
        else{
            return -1;
        }
      }
      return chm;
    }
}